Adding coe-cli assets to pipelines.

mikefactorial opened the pull request:
Fixes microsoft/coe-starter-kit#3523

devkeydet commented:
Instead of calling coe-linux and coe-win, please consider the pattern the actions team takes with pac?  They put the different executables in different directories per platform.
https://github.com/microsoft/powerplatform-actions/tree/main/dist

Also, I still wonder what confusion we may be causing by distributing a linux executable when our pipelines still don't work with linux agents.  Perhaps add a readme to explain?

The **_coe-win.exe --version_** command returns 0.0.1.  Ideally, we'd formally version the cli consistently with the rest of our releases.  Given the time sensitivity of the release we want to include this in, it's ok to keep the version for this PR.  However, let's consider formal versioning in the next release as well as making the exe part of our releases moving forward.

devkeydet reviewed (changes requested):
See my comments on the PR

mikefactorial commented:
> Instead of calling the exes coe-linux and coe-win.exe, please consider the pattern the actions team takes with pac? They put the different executables (same name) in different directories per platform. https://github.com/microsoft/powerplatform-actions/tree/main/dist
> 
> Also, I still wonder what confusion we may be causing by distributing a linux executable when our pipelines still don't work with linux agents. Perhaps add a readme to explain?
> 
> The coe-win.exe --version command returns 0.0.1. Ideally, we'd formally version the cli consistently with the rest of our releases. Given the time sensitivity of the release we want to include this in, it's ok to keep the version for this PR. However, let's consider formal versioning in the next release as well as making the exe part of our releases moving forward.

I've moved the linux executable to a subdirectory similar to pac and updated the version. Will incorporate this into build pipelines. I think leaving the linux executable and pipeline logic for eventual linux support can be useful to others who may want to use our pipelines as a reference even if the specific code won't be executed in the context of our full pipeline as of today.
